src/internal/ppsutil/decoder.go

```diff
@@ -127,3 +131,182 @@ func (r *PipelineManifestReader) NextCreatePipelineRequest() (*ppsclient.CreateP
 		return r.convertRequest(document)
 	}
 }
+
+// A SpecReader reads JSON or YAML specs.  There may be a single spec, a series
+// of YAML documents and/or an array of specs.  This behavior is copied from
+// PipelineManifestReader.
+type SpecReader struct {
+	decoder *yaml.Decoder
+	next    func() (string, error)
+}
+
+// NewSpecReader returns a SpecReader which reads from r.
+func NewSpecReader(r io.Reader) *SpecReader {
+	return &SpecReader{decoder: yaml.NewDecoder(r)}
+}
+
+// Next returns the next pipeline spec as a JSON string.  If there are multiple
+// YAML documents, it will return them one at a time.  If there is an array of
+// specs, it will return them one at a time.
+func (r *SpecReader) Next() (string, error) {
+	if r.next != nil {
+		result, err := r.next()
+		switch {
+		case errors.Is(err, io.EOF):
```

Comment:
It is dangerous to run `errors.Is` without checking for nil first.  Here is a playground that demonstrates this: https://go.dev/play/p/dVK0okSjzR5